VContainer/Assets/VContainer/Runtime/Container.cs

```diff
@@ -117,14 +118,42 @@ object CreateTrackedInstance(IRegistration registration)
         IRegistration FindRegistration(Type type)
         {
             IScopedObjectResolver scope = this;
+            
+            CollectionRegistration collectionRegistration = null;
             while (scope != null)
             {
                 if (scope.TryGetRegistration(type, out var registration))
                 {
-                    return registration;
+                    if (registration is CollectionRegistration collection)
+                    {
+                        var elementType = type.GetGenericArguments()[0];
+                        if (collectionRegistration == null)
+                        {
+                            collectionRegistration = new CollectionRegistration(elementType);
+                        }
+                        if (AnnotationUtility.IsLifecycleAnnotation(elementType))
+                        {
+                            return registration;
+                        }
+                        
+                        foreach (var registration1 in collection)
```

Comment:
Couldn't you shorten the processing of this method a bit by adding some functionality to CollectionRegistration itself?

Also, and this is my mistake, but CollectionRegistration's GetEnumerator() causes a boxing allocation.